Fix for Heishamon v3 and added Thermostates

jgaalen opened the pull request:
Fixed parse issue since Heishamon v3
Created thermostates for Curves and Delta temperatures which can be set
Created thermostate for Max_Pump_Duty to be able to change the value
Update domoticz values only if they really update. This way, you can validate the lasteUpdated state in events to make sure the new value is actually different (if Threeway Valve changes from Room to DHW for example)
Fixed Holiday State and Force DHW State switches to be properly set as selector switch or normal switch

Commit:
Fixed parse issue since Heishamon v3
Created thermostates for Curves and Delta temperatures which can be set
Created thermostate for Max_Pump_Duty to be able to change the value

Commit:
Fixed parse issue since Heishamon v3
Created thermostates for Curves and Delta temperatures which can be set
Created thermostate for Max_Pump_Duty to be able to change the value

Commit:
Only update domoticz value if value changes

Commit:
Only update domoticz value if value changes
Fixed Force DHW State and Holiday state switches

Commit:
Only update domoticz value if value changes
Fixed Force DHW State and Holiday state switches

Commit:
Only update domoticz value if value changes
Fixed Force DHW State and Holiday state switches

MikeyMan83 commented:
Any way you also had a look at the Power and Cop graphs? 

Can you share the updated files

I can view the changes, but cannot download the fixed .py files if i'm correct?

jgaalen commented:
> 

It isn't merged to main, so if you want to use the update, you need to clone my fork: https://github.com/jgaalen/HeishamonMQTT.git

What is wrong with the Power and Cop graphs?

MikeyMan83 commented:
Current value is showing up nicely. But the historical graphs are not I'm afraid.

MikeyMan83 commented:
@jgaalen This is what i see for instance for the Heat_energy_production:

![image](https://user-images.githubusercontent.com/6270724/214514626-d6524394-626f-43cb-a6d1-4365fd8508e9.png)


![image](https://user-images.githubusercontent.com/6270724/214514492-6620ee7c-dfa2-4bd2-8783-70989456e8a0.png)

jgaalen commented:
@MikeyMan83 I see the issue. It's because Heishamon only reports the production in watts, not in kWh. If there is no kWh measurement, it just reports 0.
COP graphs are fine right?

MikeyMan83 commented:
Ah ok, check... Any way around it in the plugin itself?

Op wo 25 jan. 2023 om 10:16 schreef jgaalen ***@***.***>:

> @MikeyMan83 <https://github.com/MikeyMan83> I see the issue. It's because
> Heishamon only reports the production in watts, not in kWh. If there is no
> kWh measurement, it just reports 0
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MarFanNL/HeishamonMQTT/pull/13#issuecomment-1403303674>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABP26BFE73VCTKP2WYKVKDDWUDVNVANCNFSM6AAAAAATKGRKYU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

MikeyMan83 commented:
Not doubting your knowledge btw, but i tried to fill Domoticz through MQTT / Node red directly.

And somehow, even with that value being in watts as well it fills:

![image](https://user-images.githubusercontent.com/6270724/214527313-67c127e1-87ca-4c95-9c3d-ee7968ee48db.png)

![image](https://user-images.githubusercontent.com/6270724/214527455-a3151775-08b8-44d6-a5f1-91f651ceacc1.png)

Also seems to be watts... Now i'm confused...

jgaalen commented:
> Not doubting your knowledge btw, but i tried to fill Domoticz through MQTT / Node red directly.
> 
> And somehow, even with that value being in watts as well it fills:
> 
> ![image](https://user-images.githubusercontent.com/6270724/214527313-67c127e1-87ca-4c95-9c3d-ee7968ee48db.png)
> 
> ![image](https://user-images.githubusercontent.com/6270724/214527455-a3151775-08b8-44d6-a5f1-91f651ceacc1.png)
> 
> Also seems to be watts... Now i'm confused...

It's probably because the mqtt default plugin stores it as a 'watt device', as you can see by the type ('Usage') and the heishamonmqtt plugin save it as a 'kwh' device, but then stores the 'watt' value and the counter explicitly as '0'

MikeyMan83 commented:
Ah ok... Clear!

Seems fixable ;)

MikeyMan83 commented:
Looking at the plugin, it seems that all devices are created as kWh devices:

`kWh_devices =["Cool_Energy_Consumption", "Cool_Energy_Production", "DHW_Energy_Consumption", "DHW_Energy_Production", "Heat_Energy_Consumption", "Heat_Energy_Production"]`

This should be 'Energy devices' i think? Not easy to change that around in a running situation i assume?

Looking further on https://www.domoticz.com/wiki/Developing_a_Python_plugin#Available_Device_Types.

Current type is Type 243 General with subtype 29 Kwh.
This should be Type 248 Usage with subtype 1 Electric

jgaalen commented:
I'll try to change it. No problem doing it for a running situation, just remove the device and at the next poll request it will be recreated. It will have a new idx and no history tho, but the history is '0' anyway

Commit:
Changed kWh devices to Watt devices and recreate the devices
Only update thermostat devices on every poll
Fix counters if there's a null value

MikeyMan83 commented:
Wow thats quick!
Testing now 👍 

No errors so far, and devices recreated nicely.

jgaalen commented:
Yes, I've tested it. It recreated the devices if they're old. It will now graph in domoticz as well. Too bad my influxdb/grafana is messed up as the measurement got from an 'Instant' to 'Usage'. Influxdb queries can't merge that nicely, so have to move the values. 

But at least the device in domoticz seems to be right and domoticz graphs itself will be there. Although, not sure how valuable the graphs are. It will just copy the value every 5 minutes to the history. All values between those 5 minutes aren't averaged or something, so the graph won't be accurate

MikeyMan83 commented:
And confirmation here as well! :)

![image](https://user-images.githubusercontent.com/6270724/214792798-c6745791-cada-4774-a7ae-a94be6ecfe85.png)

That 5 minute interval is down to the energy device in Domoticz?

MikeyMan83 commented:
@IgorYbema I think we can merge the fixes from @jgaalen

Commit:
Added SetPump

JWGracht commented:
New Heishamon version requires to change all _Energy_ to _Power_
